File: DesignPatternsExercises/java/FactoryMethodDesignPattern/Solution/PostCodeFactory.java
public class PostCodeFactory {
    public static PostCode getInstance(Country country, String postalCode) {
        if (country == Country.US)
            // return new __________(4)__________;
            return new USPostCode(postalCode);
        if (country == Country.UK)
            // return new __________(5)__________;
            return new UKPostCode(postalCode);
        if (country == Country.IN)
            // return new __________(6)__________;
            return new INPostCode(postalCode);

        return null;
    }
}

File: DesignPatternsExercises/ruby/factory_method_design_pattern/exercise/post_code_factory.rb
require_relative 'us_post_code'
require_relative 'uk_post_code'
require_relative 'in_post_code'

class PostCodeFactory
  def __________(3)__________(country, post_code)
    return __________(4)__________(post_code) if country == :US
    return __________(5)__________(post_code) if country == :UK
    return __________(6)__________(post_code) if country == :IN

    raise NotImplementedError, 'Country is not supported'
  end
end
